src/Symfony/Bundle/FrameworkBundle/Tests/Controller/RedirectControllerTest.php

```diff
@@ -235,6 +235,25 @@ public function testPathQueryParams($expectedUrl, $path, $queryString)
         $this->assertRedirectUrl($returnValue, $expectedUrl);
     }
 
+    public function testRedirectWithQuery()
+    {
+        $scheme = 'http';
+        $host = 'www.example.com';
+        $baseUrl = '/base';
+        $port = 80;
+
+        $request = $this->createRequestObject($scheme, $host, $port, $baseUrl, 'base=zaza');
+        $request->query = new ParameterBag(array('base' => 'zaza'));
+        $request->attributes = new ParameterBag(array('_route_params' => array()));
+        $urlGenerator = $this->getMockBuilder(UrlGeneratorInterface::class)->getMock();
+        $urlGenerator->expects($this->any())->method('generate')->will($this->returnValue('/test?base=zaza'));
```

Comment:
this does not test anything, as you don't assert arguments received by the router. Try to run your test on the unpatched codebase of the controller, and you will see that it will pass too.